lib/routes/pos/home/pos_invoice.dart

```diff
@@ -390,56 +434,87 @@ class POSInvoiceState extends State<POSInvoice> {
   onAddition() {
     setState(() {
       _totalAmount += _currentAmount;
-      _chargeAmountController.text = _currency.format((Int64(_totalAmount)),
-          fixedDecimals: true, includeSymbol: true);
-      _currentAmount = 0;
-      _amountController.text = _currency.format((Int64(_currentAmount)),
-          fixedDecimals: true, includeSymbol: false);
+      _clearCurrentAmounts();
+      _updateAmountControllers();
     });
   }
 
   onNumButtonPressed(String numberText) {
     setState(() {
-      _currentAmount = (_currentAmount * 10) + int.parse(numberText);
-      _amountController.text = _currency.format((Int64(_currentAmount)),
-          fixedDecimals: true, includeSymbol: false);
-      _chargeAmountController.text = _currency.format(
-          (Int64(_totalAmount + _currentAmount)),
-          fixedDecimals: true,
-          includeSymbol: true);
+      if (!_usingFiat) {
+        _currentAmount = (_currentAmount * 10) + int.parse(numberText);
+      } else {
+        _currentFiatAmount = _currentFiatAmount * 10 +
+            int.parse(numberText) /
+                pow(10, _fiatCurrencyConversion.currencyData.fractionSize);
+        _currentAmount =
+            _fiatCurrencyConversion.fiatToSat(_currentFiatAmount).toInt();
+      }
+      _updateAmountControllers();
     });
   }
 
   changeCurrency(value) {
     setState(() {
-      _currency = Currency.fromSymbol(value);
-      _userProfileBloc.currencySink
-          .add(Currency.currencies[Currency.currencies.indexOf(_currency)]);
+      bool usingFiat = _fiatCurrencyConversion != null ? true : false;
+      Currency.currencies.forEach((currency) {
```

Comment:
If we pass the symbol from the dropdown, we don't have to iterate all.

Reply:
[Done](https://github.com/breez/breezmobile/pull/268/commits/d0ffcf54a357ddac6ea495a979e616f2024361df)